Modules/getpath.py

```diff
@@ -612,6 +619,9 @@ def search_up(prefix, *landmarks, test=isfile):
             exec_prefix = prefix
         if not exec_prefix and executable_dir:
             exec_prefix = search_up(executable_dir, PLATSTDLIB_LANDMARK, test=isdir)
+        if not exec_prefix and base_executable:
+            base_executable_dir = dirname(base_executable)
+            exec_prefix = search_up(base_executable_dir, PLATSTDLIB_LANDMARK, test=isdir)
```

Comment:
And here:

- Change the check to `pyvenvcfg_home and not exec_prefix and base_executable`
- Add a comment before the check, also referencing `GH-127440`
- Add empty lines above and below the section, to make it clear what code the comment is referring to